newsroom/section_filters/views.py

```diff
@@ -1,86 +1,123 @@
 import re
+import json
+from typing import Any, Dict, Optional
 
 from bson import ObjectId
+from quart import abort
 from quart_babel import gettext
+from pydantic import BaseModel, field_validator
 
-from superdesk.core import get_current_app
-from superdesk.flask import jsonify, request
-from superdesk import get_resource_service
+from superdesk.core.web import EndpointGroup, Response, Request
 
 from newsroom.decorator import admin_only
-from newsroom.section_filters import blueprint
+from newsroom.core import get_current_wsgi_app
 from newsroom.utils import (
+    create_or_abort,
     get_json_or_400,
-    get_entity_or_404,
-    set_original_creator,
-    set_version_creator,
+    success_response,
 )
-from newsroom.utils import query_resource
 
+from .service import SectionFiltersService
+from .model import SectionFilter
 
-def get_settings_data():
+section_filters_endpoints = EndpointGroup("section_filters", __name__)
+
+
+async def get_settings_data():
     """Get the settings data for section filter
 
     :param context
     """
+    all_filters = [obj async for obj in SectionFiltersService().get_all_raw()]
+
     data = {
-        "section_filters": list(query_resource("section_filters")),
-        "sections": get_current_app().as_any().sections,
+        "section_filters": all_filters,
+        "sections": get_current_wsgi_app().sections,
     }
+
     return data
 
 
-@blueprint.route("/section_filters", methods=["GET"])
+def validate_section_filter(section_filter: dict) -> Response | None:
```

Comment:
This type of validation should now be covered by the Pydantic model validation

Reply:
that's right